core/builder.go

```diff
@@ -130,7 +130,16 @@ func setupNode(ctx context.Context, n *IpfsNode, cfg *BuildCfg) error {
 	}
 
 	var err error
-	n.Blockstore, err = bstore.WriteCached(bstore.NewBlockstore(n.Repo.Datastore()), kSizeBlockstoreWriteCache)
+	bs := bstore.NewBlockstore(n.Repo.Datastore())
+	n.Blockstore, err = bstore.WriteCached(bs, kSizeBlockstoreWriteCache)
+
+	rcfg, err := n.Repo.Config()
```

Comment:
I can't comment on non-changed lines, so here: the other `n.Repo.Config()` in this function can probably be removed.

The error handling is a bit off here too. You're throwing away the error from `bstore.WriteCached()`, and you can omit the `== nil` check if you just move that thing to *after* the usual `!= nil` check.